src/activity.rs

```diff
@@ -74,9 +74,26 @@ pub struct Payment {
 #[derive(PartialEq, Debug)]
 pub enum Recipient {
     LightningAddress { address: String },
+    LnUrlPayDomain { domain: String },
     Unknown,
 }
 
+impl Recipient {
+    pub(crate) fn new(payment_details: &LnPaymentDetails) -> Recipient {
```

Comment:
I thought about that and was even going to implement the `From` trait, but I decided against it because `From` is intended for conversions that don't incur information loss. IMO a constructor is more appropriate here.